docs/guides/middleware.md

```diff
@@ -0,0 +1,72 @@
+# Middleware
+
+Middleware is a Video.js feature that allows interaction with and modification of how the `Player` and `Tech` talk to each other. For more in-depth information, check out our [feature spotlight](http://blog.videojs.com/feature-spotlight-middleware/).
+
+## Table of Contents
+
+* [Understanding Middleware](#understanding-middleware)
+* [Using Middleware](#using-middleware)
+* [setSource](#setsource)
+
+## Understanding Middleware
+
+Middleware are functions that return an object with methods matching those on the `Tech`. There are currently a limited set of allowed methods that will be understood by middleware, and can be referenced in the [middleware source](https://github.com/videojs/video.js/blob/master/src/js/tech/middleware.js).
```

Comment:
I think we should just list the allowed ones here.

Reply:
I suppose the allowed methods won't be updated that frequently that it becomes difficult to maintain 🤔 Sounds good

Reply:
yeah, I'd hope that the next time this is updated, we just open it fully.

Reply:
Updated!